src/puppetlabs/trapperkeeper/services/scheduler/scheduler_service.clj

```diff
@@ -30,27 +57,49 @@
     (log/debug "Initializing Scheduler Service")
     (let [pool (core/create-pool)]
       (assoc context :pool pool
-                     :jobs (atom #{}))))
+                     :jobs (atom #{})
+                     :after-id (atom 0))))
 
   (stop [this context]
     (log/debug "Shutting down Scheduler Service")
     ; Stop any jobs that are still running
     (core/stop-all-jobs! @(:jobs context) (get-pool this))
+    (reset! (:jobs context) #{})
```

Comment:
Now that we have "reset" behavior with tk, it looked like the job identifiers wouldn't get cleaned up properly on stop (if they were left running), which could result in a memory leak.